src/composition.rs

```diff
@@ -403,23 +365,44 @@ impl<'a> ast::walk::Visitor<'a> for CompositionQueryAnalyzer {
                 if let ast::Expression::Identifier(identifier) =
                     &call_expr.callee
                 {
-                    if identifier.name.as_str() == "from" {
-                        if let ast::Expression::Object(object_expr) =
-                            &call_expr.arguments[0]
-                        {
-                            let ast::Property {
-                                base: _,
-                                key: _,
-                                separator: _,
-                                value,
-                                comma: _,
-                            } = &object_expr.properties[0];
-                            if let Some(ast::Expression::StringLit(
-                                ast::StringLit { base: _, value },
-                            )) = value
+                    match identifier.name.as_str() {
+                        "from" => {
+                            if let ast::Expression::Object(
+                                object_expr,
+                            ) = &call_expr.arguments[0]
                             {
-                                self.bucket = value.clone()
+                                let ast::Property {
+                                    base: _,
+                                    key: _,
+                                    separator: _,
+                                    value,
+                                    comma: _,
+                                } = &object_expr.properties[0];
+                                if let Some(
+                                    ast::Expression::StringLit(
+                                        ast::StringLit {
+                                            base: _,
+                                            value,
+                                        },
+                                    ),
+                                ) = value
+                                {
+                                    self.bucket = value.clone()
+                                }
                             }
+                            return false;
+                        }
+                        "range" => return false,
```

Comment:
So user modified ranges (e.g. `|> range(start: -12m, stop: now())`) will not be preserved. Correct?

Reply:
Correct. We had to draw lines for "this is where the end of this patch is" at some point. If we implemented everything in the same patch, we'd be blocked on other work that needs to occur and attempting to review a 5k+ line patch instead of a +2k line patch. This doesn't take away from the existing functionality, so I felt it a good stopping point.